### PR TITLE
clojure-lsp: 2021.06.01-16.19.44 -> 2021.07.12-12.30.59

### PR DESCRIPTION
**Pull request by ericdallo**
###### Motivation for this change
Bump clojure-lsp to latest release

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/clojure-lsp/default.nix [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
     url = "https://github.com/clojure-lsp/clojure-lsp/releases/download/${version}/clojure-lsp.jar";
-    sha256 = "sha256-V12rSYv/Yu12ZpLSROd+4pyGiEGRfJ7lmRqCeikcQ5Q=";
+    sha256 = "02k1k0slh1lm7k43d52jvgl0fdyp9gcr8csbr6yi71qbhy0axrmp";
   };
 
   GRAALVM_HOME = graalvm11-ce;
   CLOJURE_LSP_JAR = jar;
   CLOJURE_LSP_XMX = "-J-Xmx4g";
 
-  patches = lib.optionals stdenv.isDarwin [ ./disable-check-toolchain.patch ];
```
Comment (ericdallo) [inline pkgs/development/tools/misc/clojure-lsp/default.nix:23]: Removed this as it's already on this release

File: pkgs/development/tools/misc/clojure-lsp/default.nix [hunk 2]
```diff
@@ -1,28 +1,26 @@
...
-
-  buildInputs = [ graalvm11-ce ];
+  buildInputs = [ graalvm11-ce clojure ];
```
Comment (ericdallo) [inline pkgs/development/tools/misc/clojure-lsp/default.nix:23]: The api tests now need clojure installed

**Review by thiagokokada — APPROVED** [review]

**Comment by babariviere** [comment]
Result of `nixpkgs-review pr 130038` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clojure-lsp</li>
  </ul>
</details>

**Review by babariviere — APPROVED** [review]

**Comment by ericdallo** [comment]
@SuperSandro2000 Could you merge this, please?

**Pull request merged**